### PR TITLE
Partially resolve #23

### LINKED ISSUE #23: Fix the fight strategy
Hey, I get it, I have the same feeling towards Google about this as you do. Their idea for the web is even more deranged than FLoCs.

But, just like a friend of mine said, I think it would be better to properly tell the user about the issue without denying the user access to the website. Blocking access is what Google plans to do for anyone not meeting their unreasonable DRM mechanism requirements, so using that same technique against a company with the power of the market share will do us no good.

I suggest using an alert or whatever else is applicable to briefly explain to the user the situation that Google is creating with the Web Integrity Environment and explain the implications behind it.

### PR DESCRIPTION
**Pull request by ghost**
Translation still needs to be added. I know the navigator.language etc. is duplicated but the context that the onclick event runs in doesn't have access to the already defined variable

### CODE REVIEW

**Comment by ghost** [comment]
The commit description has a mix up of words but I don't know how to change it, the PR description is fine

**Review by Young-Lord — DISMISSED** [review]
you can store `lang` to `window.FightForTheOpenWeb` as you have already declared it to `window`.

**Comment by Young-Lord** [comment]
btw, shall we split translation to a JSON file, and using a `meta` dict to specify the Firefox download url?

**Comment by ghost** [comment]
Sure, sounds good

Splitting the translation into the JSON file should probably be done as part of #7, right?

File: openweb.js [hunk 1]
```diff
@@ -1,118 +1,219 @@
...
+            // For some reason, I can't get a normal :hover working with CSS, but I want a hover effect on this button.
+            // As such, I'm going to use JS for that.
+            document.getElementById("dismissButtonContainer").innerHTML = `
+              <button
+                onmouseenter="this.style.background='#e63939'"
+                onmouseleave="this.style.background='#d62f2f'"
+                onclick="window.FightForTheOpenWeb.dismiss()"
+                style="background: #d62f2f !important;"
+              >
+                  Continue anyways
```
Comment (Young-Lord) [inline openweb.js:204]: use `createElement` and `appendChild` for more flexibility.

File: openweb.js [hunk 2]
```diff
@@ -1,118 +1,219 @@
...
+        // fuck it
+        const text = (translations[lang] ??
+            translations[lang.split("-")[0]] ??
+            translations["en-us"])["message"];
+
+        // Just in case the website has overwritten some of these styles, we're going to set everything as !important and overwrite things like `display` and `position` that could end up screwing up the text.
+        // We're also setting the z-index value in case something gets added to the body after this script runs.
+
+        var alertDivToAdd = document.createElement("div");
+        alertDivToAdd.id = "FightForTheOpenWebDiv";
```
Comment (Young-Lord) [inline openweb.js:109]: make these strings constants using `const xxx = "Fightxxx"`.

File: openweb.js [hunk 3]
```diff
@@ -1,118 +1,219 @@
...
-    // Just in case the website has overwritten some of these styles, we're going to set everything as !important and overwrite things like `display` and `position` that could end up screwing up the text.
-    // We're also setting the z-index value in case something gets added to the body after this script runs.
+    /**
+     * Dismisses the popup. Used in the script, but can also be used by the website owner for whatever purposes they like.
+     * @returns nothing
+     */
+    window.FightForTheOpenWeb.dismiss = () => {
+        if (!window.FightForTheOpenWeb.isDismissed()) {
+            localStorage.setItem("FightForTheOpenWebDismiss", "yes");
+            document.getElementById("FightForTheOpenWebDiv").remove();
```
Comment (Young-Lord) [inline openweb.js:41]: Swap line 41 and 42 in case localStorage does not work.

**Review by Young-Lord — CHANGES_REQUESTED** [review]

**Comment by Young-Lord** [comment]
Also, you can split translations in this pr if it's convenient for you.

**Pull request closed without merging**